common/pinot/pinotQueryValidator.go

```
@@ -120,6 +127,14 @@ func (qv *VisibilityQueryValidator) validateRangeExpr(expr sqlparser.Expr) (stri
 	}
 
 	if definition.IsSystemIndexedKey(colNameStr) {
+		if _, ok = timeSystemKeys[colNameStr]; ok {
+			if lowerBound, ok := rangeCond.From.(*sqlparser.SQLVal); ok {
+				rangeCond.From, _ = trimTimefieldValueFromNanoToMilliSeconds(lowerBound)
```

let's not ignore errors. return err from this func when trimming fails.